scripts/reportInstalls.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { promisify } from 'util';
+import { exec } from 'child_process';
+import * as appInsights from 'applicationinsights';
+import { DEFAULT_AIKEY } from '../packages/salesforcedx-utils-vscode/src/constants';
+
+const promisifiedExec = promisify(exec);
+
+interface InstallTelemetryData {
+  // To align with the properties property for appInsightsClient.trackEvent
+  [key: string]: string;
```

Comment:
kinda weird, but this if the type required by appinsights method call.  It at least ensures all the property values are strings.